controllers/core/event_controller.go

```diff
@@ -0,0 +1,235 @@
+			Help: "Event controller reconcile events latency in ms",
+		},
+		[]string{"time"},
+	)
+
+	prometheusRegistered = false
+	reconcileEvents      = "reconcile_events"
+)
+
+func PrometheusRegister() {
+	if !prometheusRegistered {
+		metrics.Registry.MustRegister(eventControllerTotalEventsCount)
+		metrics.Registry.MustRegister(eventControllerSGPEventsCount)
+		metrics.Registry.MustRegister(eventControllerEventFailureCount)
+		metrics.Registry.MustRegister(eventControllerCNEventsCount)
+		metrics.Registry.MustRegister(eventReconcileOperationLatency)
+		metrics.Registry.MustRegister(eventControllerSGPEventsFailureCount)
+		metrics.Registry.MustRegister(eventControllerCNEventsFailureCount)
+		prometheusRegistered = true
+	}
+}
+
+func (r *EventReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	start := time.Now()
+	var err error
+
+	logger := r.Log.WithValues("event", req.Name)
+
+	eventControllerTotalEventsCount.WithLabelValues("reconcile").Inc()
+
+	ops := []client.ListOption{
+		client.MatchingFields{
+			EventFilterKey: config.VpcCNIReportingAgent,
+		},
+	}
+
+	events, err := r.K8sAPI.ListEvents(ops)
+	if err != nil {
+		logger.Error(err, "List events failed")
+		eventControllerEventFailureCount.WithLabelValues("failing_list").Inc()
+		return ctrl.Result{}, err
+	}
+
+	for _, event := range events.Items {
+		// only check and process events were created in the past EventCreatedPastMinutes minutes
+		if !event.GetCreationTimestamp().After(time.Now().Add(-time.Minute * EventCreatedPastMinutes)) {
+			continue
+		}
+		if r.isValidEventForSGP(event) || r.isValidEventForCustomNetworking(event) {
+			nodeName := string(event.InvolvedObject.UID)
+
+			if node, err := r.K8sAPI.GetNode(nodeName); err != nil {
+				// if not found the node, we don't requeue and just wait VPC CNI to send another request
+				r.Log.V(1).Info("Event reconciler didn't find the node and wait for next request for the node", "Node", node.Name)
+				return ctrl.Result{}, nil
+			} else {
+				if r.isEventToManageNode(event) && r.isValidEventForSGP(event) {
+					if err = r.K8sAPI.AddLabelToManageNode(node, config.HasTrunkAttachedLabel, "true"); err != nil {
+						// by returning an error, we request that our controller will get Reconcile() called again
+						// we use the GetNode() as a safeguard to avoid repeating reconciling on deleted nodes
+						eventControllerSGPEventsFailureCount.WithLabelValues("trunk_label_failed").Inc()
+						return ctrl.Result{}, err
+					}
+					eventControllerSGPEventsCount.WithLabelValues("trunk_labelled").Inc()
+					r.Log.Info("Lable node with trunk label as true", "Node", node.Name, "Label", config.HasTrunkAttachedLabel)
+				}
+				if r.isValidEventForCustomNetworking(event) {
+					configKey, configName := parseEventMsg(event.Message)
+					if configKey == config.CustomNetworkingLabel {
+						if err = r.K8sAPI.AddLabelToManageNode(node, config.CustomNetworkingLabel, configName); err != nil {
```

Comment:
If the trunk label succeeded but not the custom networking label, based on your comment reconciler should pick the event again and we will again try to add the trunk label in line 158 isn't it? wondering if client-go allows relabelling?

Reply:
@jayanthvn same as the above comment, the `AddLabelToManagedNode` will not relabel the node object if the label matches requested. Requeued requests should be safe if one of them fails in the previous attempt.